flood_adapt/integrator/sfincs_adapter.py

```diff
@@ -426,58 +807,120 @@ def add_spw_forcing(
+        elif mode == Mode.risk:  # risk mode requires an additional folder layer
+            for subevent in event.get_subevents():
+                simulation_paths.append(
+                    results_path.joinpath(
+                        "simulations",
+                        subevent.attrs.name,
+                        self._site.attrs.sfincs.overland_model,
+                    )
+                )
 
-    def add_obs_points(self):
-        """Add observation points provided in the site toml to SFINCS model."""
-        if self.site.attrs.obs_point is not None:
-            obs_points = self.site.attrs.obs_point
-            names = []
-            lat = []
-            lon = []
-            for pt in obs_points:
-                names.append(pt.name)
-                lat.append(pt.lat)
-                lon.append(pt.lon)
+                # Create a folder name for the offshore model (will not be used if offshore model is not created)
+                simulation_paths_offshore.append(
+                    results_path.joinpath(
+                        "simulations",
+                        subevent.attrs.name,
+                        self._site.attrs.sfincs.offshore_model,
+                    )
+                )
 
-            # create GeoDataFrame from obs_points in site file
-            df = pd.DataFrame({"name": names})
-            gdf = gpd.GeoDataFrame(
-                df, geometry=gpd.points_from_xy(lon, lat), crs="EPSG:4326"
-            )
+        return simulation_paths, simulation_paths_offshore
 
-            # Add locations to SFINCS file
-            self.sf_model.setup_observation_points(locations=gdf, merge=False)
+    def _get_flood_map_path(self) -> list[Path]:
+        """_summary_."""
+        results_path = self._get_result_path()
+        mode = Database().events.get(self._scenario.attrs.event).get_mode()
+
+        if mode == Mode.single_event:
+            map_fn = [results_path.joinpath("max_water_level_map.nc")]
 
-    def read_zsmax(self):
+        elif mode == Mode.risk:
+            map_fn = []
+            for rp in self._site.attrs.risk.return_periods:
+                map_fn.append(results_path.joinpath(f"RP_{rp:04d}_maps.nc"))
+
+        return map_fn
+
+    def _get_wl_df_from_offshore_his_results(self) -> pd.DataFrame:
+        """Create a pd.Dataframe with waterlevels from the offshore model at the bnd locations of the overland model.
+
+        Returns
+        -------
+        wl_df: pd.DataFrame
+            time series of water level.
+        """
+        ds_his = utils.read_sfincs_his_results(
+            Path(self._model.root).joinpath("sfincs_his.nc"),
+            crs=self._model.crs.to_epsg(),
+        )
+        wl_df = pd.DataFrame(
+            data=ds_his.point_zs.to_numpy(),
+            index=ds_his.time.to_numpy(),
+            columns=np.arange(1, ds_his.point_zs.to_numpy().shape[1] + 1, 1),
+        )
+        return wl_df
+
+    def _get_zsmax(self):
```

Comment:
that should be called from the HazardOutput class, correct?

Reply:
Well, to be more precise: in my opinion, the best would be to have this function be called as a part of the postprocess() function in the sfincsadapter, which converts the generated df to a HazardOutput object, which can then be written to disk using HazardOutput class functions to ensure formatting is the same across the board. 

Models that want to use that file then read it with HazardOutput.read_file(path) or something similar. Then Adapters will only be coupled to a small io class instead of other adapters (mainly for impact adapters this is an issue).